-- 2024-03-26 21:27:14.7208 ERROR[145] | ICHI_API.Service.SupplierService.Update. 
      MSG_CONTENT:Microsoft.EntityFrameworkCore.DbUpdateConcurrencyException: Attempted to update or delete an entity that does not exist in the store.
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryTable`1.Update(IUpdateEntry entry, IDiagnosticsLogger`1 updateLogger)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryStore.ExecuteTransaction(IList`1 entries, IDiagnosticsLogger`1 updateLogger)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(StateManager stateManager, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.<>c.<SaveChanges>b__107_0(DbContext _, ValueTuple`2 t)
   at Microsoft.EntityFrameworkCore.Storage.NonRetryingExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges()
   at ICHI.DataAccess.Repository.UnitOfWork.Save() in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Repository\UnitOfWork.cs:line 111
   at ICHI_API.Service.SupplierService.Update(Supplier supplier, String& strMessage) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Service\SupplierService.cs:line 145
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:27:35.8850 ERROR[144] | ICHI_API.Service.SupplierService.Update. 
      MSG_CONTENT:System.InvalidOperationException: The instance of entity type 'Supplier' cannot be tracked because another instance with the same key value for {'Id'} is already being tracked. When attaching existing entities, ensure that only one entity instance with a given key value is attached. Consider using 'DbContextOptionsBuilder.EnableSensitiveDataLogging' to see the conflicting key values.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.ThrowIdentityConflict(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry, Boolean updateDuplicate)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges, Boolean modifyProperties)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState entityState, Boolean acceptChanges, Boolean modifyProperties, Nullable`1 forceStateWhenUnknownKey, Nullable`1 fallbackState)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode`1 node)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState targetState, EntityState storeGeneratedWithKeySetTargetState, Boolean forceStateWhenUnknownKey)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.SetEntityState(InternalEntityEntry entry, EntityState entityState)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Update(TEntity entity)
   at ICHI.DataAccess.Repository.SupplierRepository.Update(Supplier obj) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Repository\SupplierRepository.cs:line 23
   at ICHI_API.Service.SupplierService.Update(Supplier supplier, String& strMessage) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Service\SupplierService.cs:line 144
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:28:14.9914 ERROR[144] | ICHI_API.Service.SupplierService.Update. 
      MSG_CONTENT:System.InvalidOperationException: The instance of entity type 'Supplier' cannot be tracked because another instance with the same key value for {'Id'} is already being tracked. When attaching existing entities, ensure that only one entity instance with a given key value is attached. Consider using 'DbContextOptionsBuilder.EnableSensitiveDataLogging' to see the conflicting key values.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.ThrowIdentityConflict(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry, Boolean updateDuplicate)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges, Boolean modifyProperties)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState entityState, Boolean acceptChanges, Boolean modifyProperties, Nullable`1 forceStateWhenUnknownKey, Nullable`1 fallbackState)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode`1 node)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState targetState, EntityState storeGeneratedWithKeySetTargetState, Boolean forceStateWhenUnknownKey)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.SetEntityState(InternalEntityEntry entry, EntityState entityState)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Update(TEntity entity)
   at ICHI.DataAccess.Repository.SupplierRepository.Update(Supplier obj) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Repository\SupplierRepository.cs:line 23
   at ICHI_API.Service.SupplierService.Update(Supplier supplier, String& strMessage) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Service\SupplierService.cs:line 144
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:34:24.1221 ERROR[145] | ICHI_API.Service.SupplierService.Update. 
      MSG_CONTENT:System.InvalidOperationException: The instance of entity type 'Supplier' cannot be tracked because another instance with the same key value for {'Id'} is already being tracked. When attaching existing entities, ensure that only one entity instance with a given key value is attached. Consider using 'DbContextOptionsBuilder.EnableSensitiveDataLogging' to see the conflicting key values.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.ThrowIdentityConflict(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry, Boolean updateDuplicate)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges, Boolean modifyProperties)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState entityState, Boolean acceptChanges, Boolean modifyProperties, Nullable`1 forceStateWhenUnknownKey, Nullable`1 fallbackState)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode`1 node)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState targetState, EntityState storeGeneratedWithKeySetTargetState, Boolean forceStateWhenUnknownKey)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.SetEntityState(InternalEntityEntry entry, EntityState entityState)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Update(TEntity entity)
   at ICHI.DataAccess.Repository.SupplierRepository.Update(Supplier obj) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Repository\SupplierRepository.cs:line 23
   at ICHI_API.Service.SupplierService.Update(Supplier supplier, String& strMessage) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Service\SupplierService.cs:line 145
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:38:01.1840 ERROR[156] | ICHI_API.Service.SupplierService.Update. 
      MSG_CONTENT:System.InvalidOperationException: The instance of entity type 'Supplier' cannot be tracked because another instance with the same key value for {'Id'} is already being tracked. When attaching existing entities, ensure that only one entity instance with a given key value is attached. Consider using 'DbContextOptionsBuilder.EnableSensitiveDataLogging' to see the conflicting key values.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.ThrowIdentityConflict(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry, Boolean updateDuplicate)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges, Boolean modifyProperties)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState entityState, Boolean acceptChanges, Boolean modifyProperties, Nullable`1 forceStateWhenUnknownKey, Nullable`1 fallbackState)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode`1 node)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState targetState, EntityState storeGeneratedWithKeySetTargetState, Boolean forceStateWhenUnknownKey)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.SetEntityState(InternalEntityEntry entry, EntityState entityState)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Update(TEntity entity)
   at ICHI.DataAccess.Repository.SupplierRepository.Update(Supplier obj) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Repository\SupplierRepository.cs:line 23
   at ICHI_API.Service.SupplierService.Update(Supplier supplier, String& strMessage) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Service\SupplierService.cs:line 156
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:40:42.9811 ERROR[0] | ICHI_API.Service.SupplierService.Update. 
      MSG_CONTENT:System.InvalidOperationException: The instance of entity type 'Supplier' cannot be tracked because another instance with the same key value for {'Id'} is already being tracked. When attaching existing entities, ensure that only one entity instance with a given key value is attached. Consider using 'DbContextOptionsBuilder.EnableSensitiveDataLogging' to see the conflicting key values.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.ThrowIdentityConflict(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry, Boolean updateDuplicate)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges, Boolean modifyProperties)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState entityState, Boolean acceptChanges, Boolean modifyProperties, Nullable`1 forceStateWhenUnknownKey, Nullable`1 fallbackState)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode`1 node)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState targetState, EntityState storeGeneratedWithKeySetTargetState, Boolean forceStateWhenUnknownKey)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.SetEntityState(InternalEntityEntry entry, EntityState entityState)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Update(TEntity entity)
   at ICHI.DataAccess.Repository.SupplierRepository.Update(Supplier obj)
   at ICHI_API.Service.SupplierService.Update(Supplier supplier, String& strMessage)
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:43:55.3450 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:43:55.3681 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:43:56.3194 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:43:56.3194 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:43:56.3627 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:43:56.3627 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:43:56.3960 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:43:56.3960 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:45:20.8433 ERROR[184] | ICHI_API.Service.SupplierService.Delete. 
      MSG_CONTENT:System.InvalidOperationException: The instance of entity type 'Supplier' cannot be tracked because another instance with the same key value for {'Id'} is already being tracked. When attaching existing entities, ensure that only one entity instance with a given key value is attached. Consider using 'DbContextOptionsBuilder.EnableSensitiveDataLogging' to see the conflicting key values.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.ThrowIdentityConflict(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry, Boolean updateDuplicate)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges, Boolean modifyProperties)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState entityState, Boolean acceptChanges, Boolean modifyProperties, Nullable`1 forceStateWhenUnknownKey, Nullable`1 fallbackState)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode`1 node)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState targetState, EntityState storeGeneratedWithKeySetTargetState, Boolean forceStateWhenUnknownKey)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.SetEntityState(InternalEntityEntry entry, EntityState entityState)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Update(TEntity entity)
   at ICHI.DataAccess.Repository.SupplierRepository.Update(Supplier obj) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Repository\SupplierRepository.cs:line 23
   at ICHI_API.Service.SupplierService.Delete(Int32 id, String& strMessage) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Service\SupplierService.cs:line 184
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:54:15.5208 ERROR[121] | ICHI_API.Service.TrademarkService.Update. 
      MSG_CONTENT:System.InvalidOperationException: The instance of entity type 'Trademark' cannot be tracked because another instance with the same key value for {'Id'} is already being tracked. When attaching existing entities, ensure that only one entity instance with a given key value is attached. Consider using 'DbContextOptionsBuilder.EnableSensitiveDataLogging' to see the conflicting key values.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.ThrowIdentityConflict(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry, Boolean updateDuplicate)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges, Boolean modifyProperties)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState entityState, Boolean acceptChanges, Boolean modifyProperties, Nullable`1 forceStateWhenUnknownKey, Nullable`1 fallbackState)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode`1 node)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState targetState, EntityState storeGeneratedWithKeySetTargetState, Boolean forceStateWhenUnknownKey)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.SetEntityState(InternalEntityEntry entry, EntityState entityState)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Update(TEntity entity)
   at ICHI.DataAccess.Repository.TrademarkRepository.Update(Trademark obj) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Repository\TrademarkRepository.cs:line 23
   at ICHI_API.Service.TrademarkService.Update(Trademark trademark, String& strMessage) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Service\TrademarkService.cs:line 121
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:57:52.3827 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:57:52.4031 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:57:53.2722 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:57:53.2722 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:57:53.3167 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:57:53.3167 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:57:53.3392 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 21:57:53.3392 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:28:31.6818 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:28:31.7082 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:28:32.6475 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:28:32.6475 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:28:32.6963 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:28:32.6963 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:28:32.7264 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:28:32.7264 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:30:18.5333 INFO[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:30:18.5591 ERROR[33] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:30:19.4053 INFO[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:30:19.4053 ERROR[33] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:30:19.4513 INFO[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:30:19.4513 ERROR[33] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:30:19.4701 INFO[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:30:19.4701 ERROR[33] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:44:13.1836 ERROR[134] | ICHI_API.Service.ProductService.Create. 
      MSG_CONTENT:System.NullReferenceException: Object reference not set to an instance of an object.
   at ICHI.DataAccess.Repository.UnitOfWork.Rollback() in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Repository\UnitOfWork.cs:line 106
   at ICHI.DataAccess.Repository.UnitOfWork.Commit() in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Repository\UnitOfWork.cs:line 100
   at ICHI_API.Service.ProductService.Create(Product product, List`1 files, String& strMessage) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Service\ProductService.cs:line 134
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:51:31.7748 ERROR[0] | ICHI_API.Service.ProductService.Create. 
      MSG_CONTENT:System.InvalidOperationException: The instance of entity type 'Product' cannot be tracked because another instance with the same key value for {'Id'} is already being tracked. When attaching existing entities, ensure that only one entity instance with a given key value is attached. Consider using 'DbContextOptionsBuilder.EnableSensitiveDataLogging' to see the conflicting key values.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.ThrowIdentityConflict(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry, Boolean updateDuplicate)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges, Boolean modifyProperties)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState entityState, Boolean acceptChanges, Boolean modifyProperties, Nullable`1 forceStateWhenUnknownKey, Nullable`1 fallbackState)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode`1 node)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState targetState, EntityState storeGeneratedWithKeySetTargetState, Boolean forceStateWhenUnknownKey)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.SetEntityState(InternalEntityEntry entry, EntityState entityState)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Update(TEntity entity)
   at ICHI.DataAccess.Repository.ProductRepository.Update(Product obj)
   at ICHI_API.Service.ProductService.Create(Product product, List`1 files, String& strMessage)
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 22:55:38.2003 ERROR[167] | ICHI_API.Service.ProductService.Create. 
      MSG_CONTENT:System.NullReferenceException: Object reference not set to an instance of an object.
   at ICHI_CORE.Helpers.ImageHelper.CheckImage(IFormFile file) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Helpers\ImageHelper.cs:line 36
   at ICHI_API.Service.ProductService.Create(Product product, List`1 files, String& strMessage) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Service\ProductService.cs:line 167
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:37:02.7520 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:37:02.7821 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:37:04.1154 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:37:04.1154 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:37:04.2044 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:37:04.2044 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:37:04.2312 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:37:04.2312 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:37:04.4570 ERROR[153] | ICHI_API.Service.ProductService.Create. 
      MSG_CONTENT:System.NullReferenceException: Object reference not set to an instance of an object.
   at ICHI_API.Service.ProductService.Create(Product product, List`1 files, String& strMessage) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Service\ProductService.cs:line 153
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:38:04.9514 ERROR[153] | ICHI_API.Service.ProductService.Create. 
      MSG_CONTENT:System.NullReferenceException: Object reference not set to an instance of an object.
   at ICHI_API.Service.ProductService.Create(Product product, List`1 files, String& strMessage) in C:\Users\tien0\source\repos\Code\ICHI-E-COMMERCE\ICHI_API\Service\ProductService.cs:line 153
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:48:00.7536 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:48:00.7802 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:48:01.7315 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:48:01.7315 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:48:01.7841 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:48:01.7841 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:48:01.8306 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:48:01.8306 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:50:23.1994 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:50:23.2226 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:50:24.1703 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:50:24.1703 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:50:24.2510 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:50:24.2510 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:50:24.2791 INFO[31] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Get data from appSettings success.
      EXXCEPTION:  | MESSAGE:
-- 2024-03-26 23:50:24.2791 ERROR[32] | ICHI_TEST.ContextGenerator.GeneratorDb. 
      MSG_CONTENT:Test log error
      EXXCEPTION:  | MESSAGE:
